orion/primitives/tadgan.py

```diff
@@ -82,241 +146,261 @@ def __getstate__(self):
+        X, target = data
+        batch_size = tf.shape(X)[0]
+        mini_batch_size = batch_size // self.iterations_critic
+
+        fake = tf.ones((mini_batch_size, 1), dtype=tf.float32)
+        valid = -tf.ones((mini_batch_size, 1), dtype=tf.float32)
+
+        batch_g_loss, batch_cx_loss, batch_cz_loss = [], [], []
+
+        # Train the critics
+        for j in range(self.iterations_critic):
+            x = X[j * mini_batch_size: (j + 1) * mini_batch_size]
+            y = target[j * mini_batch_size: (j + 1) * mini_batch_size]
+            z = tf.random.normal(shape=(mini_batch_size, self.latent_shape[0],
+                                        self.latent_shape[1]), dtype=tf.float32)
+
+            with tf.GradientTape(persistent=True) as tape:
+                # Train critic x
+                x_ = self.generator(z, training=True)
+                cx_valid = self.critic_x(y, training=True)
+                cx_fake = self.critic_x(x_, training=True)
+
+                cx_valid_loss = self._wasserstein_loss(valid, cx_valid)
+                cx_fake_loss = self._wasserstein_loss(fake, cx_fake)
+                cx_gp = self._gradient_penalty_loss(y, x_, self.critic_x)
+                cx_loss = cx_valid_loss + cx_fake_loss + 10 * cx_gp
+
+                # Train critic z
+                z_ = self.encoder(x, training=True)
+                cz_valid = self.critic_z(z, training=True)
+                cz_fake = self.critic_z(z_, training=True)
+
+                cz_valid_loss = self._wasserstein_loss(valid, cz_valid)
+                cz_fake_loss = self._wasserstein_loss(fake, cz_fake)
+                cz_gp = self._gradient_penalty_loss(z, z_, self.critic_z)
+                cz_loss = cz_valid_loss + cz_fake_loss + 10 * cz_gp
+
+            # Get the gradients for the critics
+            cx_grads = tape.gradient(cx_loss, self.critic_x.trainable_weights)
+            cz_grads = tape.gradient(cz_loss, self.critic_z.trainable_weights)
+
+            # Update the weights of the critics
+            self.optimizer.apply_gradients(zip(cx_grads, self.critic_x.trainable_weights))
+            self.optimizer.apply_gradients(zip(cz_grads, self.critic_z.trainable_weights))
+
+            # Record loss
+            batch_cx_loss.append([cx_loss, cx_valid_loss, cx_fake_loss, cx_gp])
+            batch_cz_loss.append([cz_loss, cz_valid_loss, cz_fake_loss, cz_gp])
+
+        # Train encoder generator
+        with tf.GradientTape() as tape:
+            x_ = self.generator(z, training=True)
+            cx_fake = self.critic_x(x_, training=True)
+            z_ = self.encoder(x, training=True)
+            cz_fake = self.critic_z(z_, training=True)
+            x_rec_ = self.generator(z_, training=True)
+
+            # Encoder Generator Loss
+            eg_x_loss = self._wasserstein_loss(valid, cx_fake)
+            eg_z_loss = self._wasserstein_loss(valid, cz_fake)
+            eg_mse = MeanSquaredError()(x, x_rec_)
```

Comment:
Do you mean here `y` instead of `x`?

Multivariate TadGAN supports the reconstruction of a single channel.

**Example**
I want to reconstruct only the first channel
* x is shape `[100, 25]`
* y is shape `[100, 1]`

Then `generator` will have `x_rec_` of shape `[100, 1]` by definition of input and target shapes

Reply:
Yep, changed to y